dockstore-webservice/src/main/java/io/dockstore/webservice/resources/WorkflowResource.java

```diff
@@ -552,6 +563,25 @@ private void updateInfo(Workflow oldWorkflow, Workflow newWorkflow) {
         return tokenDAO.findByUserId(user.getId());
     }
 
+    /**
+     * Construct a URL for Dockstore
+     * @param dockstoreScheme Dockstore scheme, e.g. http or https
+     * @param dockstoreHostName Dockstore host name, e.g. dockstore.org
+     * @param dockstorePort Dockstore port, e.g. 4200
+     */
+    private String createHostUrl(String dockstoreScheme, String dockstoreHostName, String dockstorePort) {
+        URL url;
+        try {
+            int iport = StringUtils.isEmpty(dockstorePort) ? -1 : Integer.parseInt(dockstorePort);
+            url = new URL(dockstoreScheme, dockstoreHostName, iport, "");
+        } catch (MalformedURLException e) {
+            LOG.error("Could not create Dockstore URL."
```

Comment:
`scheme` is set once when the webservice is started and never changes.
Should just bring down the whole webservice in the constructor when an invalid scheme is provided as opposed to waiting till someone requests a DOI.  (1.7.0)

Alternatively, give the variable a proper type in the configuration file. (1.8.0)

Reply:
done